src/components/pagination/pagination.ts

```diff
@@ -0,0 +1,301 @@
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  ElementRef,
+  DynamicComponentLoader,
+  AfterViewInit,
+  AfterContentInit
+} from '@angular/core';
+import {isPresent} from '@angular/core/src/facade/lang';
+import 'rxjs/add/operator/filter';
+import {PaginationService} from './pagination_service';
+
+export interface IPaginationModel {
+  currentPage: number;
+  itemsPerPage: number;
+  totalItems: number;
+}
+
+export abstract class AbstractPaginationSubComponent {
+
+  @Input()
+  name: string;
+
+  @Input()
+  model: IPaginationModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  constructor(protected service: PaginationService) {
+    this.service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.pagination;
+      });
+  }
+
+}
+
+@Component({
+  selector: 'md-pagination-range',
+  template: '',
+  host: {
+    '[class.md-pagination-range]': 'true',
+    '[innerHTML]': 'getRange()'
+  }
+})
+export class MdPaginationRange extends AbstractPaginationSubComponent {
+
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPaginationModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  @Input('range-format')
+  rangeFormat: string = '{start}-{end} of {total}';
+
+  public value: string = '';
+
+  constructor(protected service: PaginationService) {
+    super(service);
+  }
+
+  /**
+   * tranform format into an readable string
+   *
+   * @returns {string}
+   */
+  getFormattedValue(rangeStart: number, rangeStop: number, totalItems: number) {
```

Comment:
This kind of model to display-string formatting logic seems like the perfect candidate for a [pipe](https://angular.io/docs/ts/latest/guide/pipes.html)